src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -571,16 +573,26 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerQueryHasLimit)
 		errorDetail = "Offset clause is currently unsupported";
 	}
 
-	if (subqueryTree->limitCount && !outerQueryHasLimit)
+	/*
+	 * Limit is partially supported when SubqueryPushdown is set.
+	 * The outermost query must have a limit clause.
+	 */
+	if (subqueryTree->limitCount && !SubqueryPushdown)
```

Comment:
> Hmm, shall we add a errhnt for this case? What do you think?

errhint would require exposing `citus.subquery_pushdown`. I think we never publicly mentioned that so far.